### PR TITLE
fix authentication method to return a boolean.

### PR DESCRIPTION
**Pull request by miladb**
Fix this error : no authenticationprovider found for token of class usernamepasswordtoken

### CODE REVIEW

File: Security/Authentication/Provider/UserAuthenticationProvider.php [hunk 1]
```diff
@@ -149,16 +149,15 @@ private function authenticateUser(TokenInterface $token, UserInterface $user)
...
-            }
-        } catch (Exception $e) {
-            return false;
+        $classname = \Symfony\Component\Security\Core\Util\ClassUtils::getRealClass($user);
+
+        if (true === $this->_encoderFactory
+                ->getEncoder($classname)
+                ->isPasswordValid($user->getPassword(), $token->getCredentials(), $user->getSalt())) {
+            return new UsernamePasswordToken($user, $user->getPassword(), $user->getRoles());
         }
```
Comment (eric-chau) [inline Security/Authentication/Provider/UserAuthenticationProvider.php]: Hey @miladb, thank you for your contribution! You should leave the `try-catch` and move the `return false;` out of the `catch` block. Can you do it please?

Comment (miladb) [inline Security/Authentication/Provider/UserAuthenticationProvider.php]: Hey @eric-chau. I can do it but in the catch, you want to throw the exception or just return false ?

**Comment by eric-chau** [comment]
@miladb, you don't have to do anything in the `catch` block cause we execute the same code in both case, returning `false`.

File: Security/Authentication/Provider/UserAuthenticationProvider.php (hunk 1 above)
Comment (eric-chau) [inline Security/Authentication/Provider/UserAuthenticationProvider.php]: @miladb, you don't have to do anything in the `catch` block cause we execute the same code in both case, returning `false`.

File: Security/Authentication/Provider/UserAuthenticationProvider.php [hunk 2]
```diff
@@ -152,13 +152,15 @@ private function authenticateWithEncoder(TokenInterface $token, UserInterface $u
...
             $classname = \Symfony\Component\Security\Core\Util\ClassUtils::getRealClass($user);
             if (true === $this->_encoderFactory
-                            ->getEncoder($classname)
-                            ->isPasswordValid($user->getPassword(), $token->getCredentials(), $user->getSalt())) {
+                    ->getEncoder($classname)
+                    ->isPasswordValid($user->getPassword(), $token->getCredentials(), $user->getSalt())) {
                 return new UsernamePasswordToken($user, $user->getPassword(), $user->getRoles());
             }
         } catch (Exception $e) {
             return false;
```
Comment (eric-chau) [inline Security/Authentication/Provider/UserAuthenticationProvider.php]: Sorry I might be not enough clear, but can you remove this line please? ^^ So everything is fine.

**Comment by eric-chau** [comment]
thank you @miladb!

**Pull request merged**